frappe/utils/backups.py

```diff
@@ -206,6 +206,31 @@ def get_backup():
 	recipient_list = odb.send_email()
 	frappe.msgprint(_("Download link for your backup will be emailed on the following email address: {0}").format(', '.join(recipient_list)))
 
+
+@frappe.whitelist()
+def fetch_latest_backups(with_files=True, recent=3):
+	"""Takes backup on-demand if doesnt exist satisfying the `recent` parameter
+	Only for: System Managers
+
+	Args:
+		with_files (bool, optional): If set, files will backuped up. Defaults to True.
+		recent (int, optional): Won't take a new backup if backup exists within this paramter. Defaults to 3 hours
+
+	Returns:
+		dict: relative Backup Paths
+	"""
+	frappe.only_for("System Manager")
+	odb = BackupGenerator(frappe.conf.db_name, frappe.conf.db_name, frappe.conf.db_password, db_host=frappe.db.host, db_type=frappe.conf.db_type, db_port=frappe.conf.db_port)
```

Comment:
This request will timeout for large backups.

Reply:
"fixed" in https://github.com/frappe/frappe/pull/11141